backend/packages/Upgrade/src/api/services/ExperimentUserService.ts

```diff
@@ -301,25 +347,35 @@ export class ExperimentUserService {
     });
   }
 
-  private async removeEnrollments(userId: string, groupMembership: any, oldGroupMembership: any): Promise<void> {
-    const userGroupRemovedMap: Map<string, string[]> = new Map();
-
-    // check the groups removed from setGroupMembership
-    Object.keys(oldGroupMembership).map((key) => {
-      const oldGroupArray: string[] = oldGroupMembership[key] || [];
-      const newGroupArray: string[] = (groupMembership && groupMembership[key]) || [];
-      oldGroupArray.map((groupId) => {
-        if (!(newGroupArray && newGroupArray.includes(groupId))) {
-          const groupNames = userGroupRemovedMap.has(key) ? userGroupRemovedMap.get(key) : [];
-          if (!newGroupArray) {
-            userGroupRemovedMap.set(key, [...groupNames, ...newGroupArray]);
-          } else {
-            userGroupRemovedMap.set(key, [...groupNames, groupId]);
-          }
-        }
-      });
+  /**
+   * Remove enrollments only if the working group is changed
+   * @param userId
+   * @param newWorkingGroup
+   * @param oldWorkingGroup
+   * return Promise<void>
+   */
+  private async removeEnrollments(
+    userId: string,
+    newWorkingGroup: Record<string, string>,
+    oldWorkingGroup: Record<string, string>
+  ): Promise<void> {
+    const workingGroupUpdated: string[] = [];
+
+    // check the groups removed from existing GroupMembership
+    // and populate userGroupRemoved
+    Object.entries(oldWorkingGroup).map(([key, value]) => {
+      const newWorkingGroupValue: string | undefined = newWorkingGroup[key];
+      // if the working group value has changed
+      if (newWorkingGroupValue !== value) {
+        workingGroupUpdated.push(key);
+      }
     });
 
+    // End the function if there is no change in working group
+    if (workingGroupUpdated.length === 0) {
+      return;
+    }
+
     // get all group experiments
     const groupExperiments = await this.experimentRepository.find({
       where: {
```

Comment:
Should we also not indlude 'Archived' state?

Reply:
I think we should change the syntax to just include experiments in enrolling and enrollment complete experiments.

Reply:
Yeah correct.

Reply:
code updated